### PR TITLE
Fix tests in `CryptoTransferSuite`

### LINKED ISSUE #9233: Fix tests in `CryptoTransferSuite`
Fix 

- [x] dissociatedRoyaltyCollectorsCanUseAutoAssociation 
- [x] nftTransfersCannotRepeatSerialNos
- [x] checksExpectedDecimalsForFungibleTokenTransferList
- [x] nftSelfTransfersRejectedBothInPrecheckAndHandle
- [x] hbarAndFungibleSelfTransfersRejectedBothInPrecheckAndHandle
 in `CryptoTransferSuite

Subissue of https://github.com/hashgraph/hedera-services/issues/8466

### PR DESCRIPTION
**Pull request by Neeharika-Sompalli**
Fixes #9233

### CODE REVIEW

**Review by iwsimon — APPROVED** [review]
LGTM

**Review by mhess-swl — APPROVED** [review]

**Review by iwsimon — DISMISSED** [review]
LGTM

**Review by iwsimon — DISMISSED** [review]

**Review by mhess-swl — DISMISSED** [review]

**Review by tinker-michaelj — APPROVED** [review]
So we've found an advantage of not doing excessive prechecks---simpler EETs! 😇 

Looks like we need to update [two more expectations tho](https://scans.gradle.com/s/nmc6krdji5m2c/tests/task/:test-clients:itest/details/AllIntegrationTests/concurrentSpecs()%5B1%5D?top-execution=1) - 
```
org.opentest4j.AssertionFailedError: 2 specs FAILED. By suite,	
  CryptoTransfer                   -> 2 failures:	
    HbarAndFungibleSelfTransfersRejectedBothInPrecheckAndHandle: Wrong precheck status! Expected OK, actual ACCOUNT_REPEATED_IN_ACCOUNT_AMOUNTS when executing HapiCryptoTransfer{sigs=2, node=0.0.3, transfers=[0.0.3510 <- +1, 0.0.3510 -> -1], tokenTransfers=}	
    NftSelfTransfersRejectedBothInPrecheckAndHandle: Bad answerOnlyPrecheck! expected OK, actual RECEIPT_NOT_FOUND when executing HapiGetReceipt{node=0.0.3, txn=uncheckedTxn, explicit Txn :=Optional.empty}
```

**Comment by Neeharika-Sompalli** [comment]
> So we've found an advantage of not doing excessive prechecks---simpler EETs! 😇
> 
> Looks like we need to update [two more expectations tho](https://scans.gradle.com/s/nmc6krdji5m2c/tests/task/:test-clients:itest/details/AllIntegrationTests/concurrentSpecs()%5B1%5D?top-execution=1) -
> 
> ```
> org.opentest4j.AssertionFailedError: 2 specs FAILED. By suite,	
>   CryptoTransfer                   -> 2 failures:	
>     HbarAndFungibleSelfTransfersRejectedBothInPrecheckAndHandle: Wrong precheck status! Expected OK, actual ACCOUNT_REPEATED_IN_ACCOUNT_AMOUNTS when executing HapiCryptoTransfer{sigs=2, node=0.0.3, transfers=[0.0.3510 <- +1, 0.0.3510 -> -1], tokenTransfers=}	
>     NftSelfTransfersRejectedBothInPrecheckAndHandle: Bad answerOnlyPrecheck! expected OK, actual RECEIPT_NOT_FOUND when executing HapiGetReceipt{node=0.0.3, txn=uncheckedTxn, explicit Txn :=Optional.empty}
> ```

Interesting , they are in Integration Tests which runs on mono-service ?

File: hedera-node/test-clients/src/main/java/com/hedera/services/bdd/suites/crypto/CryptoTransferSuite.java [hunk 1]
```diff
@@ -1159,16 +1158,16 @@ private HapiSpec nftSelfTransfersRejectedBothInPrecheckAndHandle() {
...
+                                .txnId(uncheckedTxn)
+                                .hasPrecheck(ACCOUNT_REPEATED_IN_ACCOUNT_AMOUNTS))
                 .then(
                         sleepFor(2_000),
-                        getReceipt(uncheckedTxn).hasPriorityStatus(ACCOUNT_REPEATED_IN_ACCOUNT_AMOUNTS),
                         getAccountInfo(owningParty).has(accountWith().noChangesFromSnapshot(owningParty)));
     }
 
+    @HapiTest
     private HapiSpec hbarAndFungibleSelfTransfersRejectedBothInPrecheckAndHandle() {
```
Comment (mhess-swl) [inline hedera-node/test-clients/src/main/java/com/hedera/services/bdd/suites/crypto/CryptoTransferSuite.java:1171]:
```suggestion
    private HapiSpec hbarAndFungibleSelfTransfersRejectedInPrecheck() {
```

File: hedera-node/test-clients/src/main/java/com/hedera/services/bdd/suites/crypto/CryptoTransferSuite.java [hunk 2]
```diff
@@ -1135,6 +1133,7 @@ private HapiSpec nftTransfersCannotRepeatSerialNos() {
                         .hasPrecheck(INVALID_ACCOUNT_AMOUNTS));
     }
 
+    @HapiTest
     private HapiSpec nftSelfTransfersRejectedBothInPrecheckAndHandle() {
```
Comment (mhess-swl) [inline hedera-node/test-clients/src/main/java/com/hedera/services/bdd/suites/crypto/CryptoTransferSuite.java:1137]:
```suggestion
    private HapiSpec nftSelfTransfersRejectedInPrecheck() {
```

**Review by mhess-swl — APPROVED** [review]

File: hedera-node/test-clients/src/main/java/com/hedera/services/bdd/suites/crypto/CryptoTransferSuite.java (hunk 1 above)
Comment (Neeharika-Sompalli) [reply hedera-node/test-clients/src/main/java/com/hedera/services/bdd/suites/crypto/CryptoTransferSuite.java:1171]: sure will do in next PR :)

File: hedera-node/test-clients/src/main/java/com/hedera/services/bdd/suites/crypto/CryptoTransferSuite.java (hunk 2 above)
Comment (Neeharika-Sompalli) [reply hedera-node/test-clients/src/main/java/com/hedera/services/bdd/suites/crypto/CryptoTransferSuite.java:1137]: Sure will do in next PR

**Pull request merged**